src/widget/paintable.cpp

```diff
@@ -0,0 +1,253 @@
+
+#include "widget/wpixmapstore.h"
+
+#include <QDir>
+#include <QString>
+#include <QtDebug>
+
+#include "util/math.h"
+#include "skin/imgloader.h"
+
+// static
+Paintable::DrawMode Paintable::DrawModeFromString(const QString& str) {
+    if (str.compare("FIXED", Qt::CaseInsensitive) == 0) {
+        return FIXED;
+    } else if (str.compare("STRETCH", Qt::CaseInsensitive) == 0) {
+        return STRETCH;
+    } else if (str.compare("STRETCH_ASPECT", Qt::CaseInsensitive) == 0) {
+        return STRETCH_ASPECT;
+    } else if (str.compare("TILE", Qt::CaseInsensitive) == 0) {
+        return TILE;
+    }
+
+    // Fall back on the implicit default from before Mixxx supported draw modes.
+    qWarning() << "Unknown DrawMode string in DrawModeFromString:"
+               << str << "using FIXED";
+    return FIXED;
+}
+
+// static
+QString Paintable::DrawModeToString(DrawMode mode) {
+    switch (mode) {
+        case FIXED:
+            return "FIXED";
+        case STRETCH:
+            return "STRETCH";
+        case STRETCH_ASPECT:
+            return "STRETCH_ASPECT";
+        case TILE:
+            return "TILE";
+    }
+    // Fall back on the implicit default from before Mixxx supported draw modes.
+    qWarning() << "Unknown DrawMode in DrawModeToString " << mode
+               << "using FIXED";
+    return "FIXED";
+}
+
+Paintable::Paintable(const PixmapSource& source, DrawMode mode, double scaleFactor)
+        : m_drawMode(mode),
+          m_source(source) {
+    if (!source.isSVG()) {
+        m_pPixmap.reset(WPixmapStore::getPixmapNoCache(source.getPath(), scaleFactor));
+    } else {
+        m_pSvg.reset(new QSvgRenderer());
+        if (source.getData().isEmpty()) {
+            m_pSvg->load(source.getPath());
+        } else {
+            m_pSvg->load(source.getData());
+        }
+        if (mode == TILE || mode == Paintable::FIXED || WPixmapStore::willCorrectColors()) {
```

Comment:
This change converts all FIXED SVGs to raster? Won't this cause blurry SVGs on hidpi displays?